src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -88,14 +102,32 @@ void PairingSession::DiscardExchange()
     }
 }
 
-CHIP_ERROR PairingSession::EncodeMRPParameters(TLV::Tag tag, const ReliableMessageProtocolConfig & mrpLocalConfig,
-                                               TLV::TLVWriter & tlvWriter)
+CHIP_ERROR PairingSession::EncodeSessionParameters(TLV::Tag tag, const Optional<ReliableMessageProtocolConfig> & providedMrpConfig,
+                                                   TLV::TLVWriter & tlvWriter)
 {
+    ReliableMessageProtocolConfig mrpLocalConfig = GetDefaultMRPConfig();
+    if (providedMrpConfig.HasValue())
+    {
+        mrpLocalConfig = providedMrpConfig.Value();
+    }
     TLV::TLVType mrpParamsContainer;
     ReturnErrorOnFailure(tlvWriter.StartContainer(tag, TLV::kTLVType_Structure, mrpParamsContainer));
     ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(1), mrpLocalConfig.mIdleRetransTimeout.count()));
     ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(2), mrpLocalConfig.mActiveRetransTimeout.count()));
     ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(3), mrpLocalConfig.mActiveThresholdTime.count()));
+
+    uint16_t dataModel = CHIP_DEVICE_DATA_MODEL_REVISION;
+    ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(4), dataModel));
+
+    uint16_t interactionModel = CHIP_DEVICE_INTERACTION_MODEL_REVISION;
+    ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(5), interactionModel));
+
+    // TODO where do I get SPECIFICATION_VERSION from? For now I have just hardcoded 1.3.
```

Comment:
Any recommendations on where I should be grabbing this spec version value?

Reply:
This needs a new header that has that value....